### DISCUSSION TITLE
invert axes when plotting 2d array

### DISCUSSION THREAD

**Discussion by saveriogzz** [pydata/xarray#5505] [Q&A] [answered]
Hi guys,

Sorry if this is a basic question but I can't find the answer.
Is there an easy way to invert the axes when plotting a 2d array?

I am calling the method `plot()` on a `xarray.core.dataarray.DataArray` with shape `(1064, 338)` but I get `time` on the y axes, while I'd like it to be on the x axes.

My workflow:
```python
import xarray as xr
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

filename = "/somewhere/something.nc"

ds = xr.open_dataset(filename)

ze2d = ds.Ze.isel()

ze2d.plot()
```
where `print(ze2d)` gives
```
<xarray.DataArray 'Ze' (time: 1064, range: 338)>
[359632 values with dtype=float32]
Coordinates:
  * time     (time) int32 636692401 636692405 636692408 ... 636695995 636695998
Dimensions without coordinates: range
Attributes:
    long_name:        Reflectivity
    comment:          vertical polarisation, linear units
    rpg_manual_name:  Ze
```

Does anyone know a convenient way to do so?

Thanks a lot

**Comment by saveriogzz** [comment]
It was as easy as calling the `transpose` method on the DataArray before plotting!

**Comment by keewis** [comment, marked as answer]
instead of using `transpose`, you can also use the `x` or `y` parameters. In your case:
```python
ze2d.plot.imshow(x="time")
```

**Reply by saveriogzz** [reply]
Thanks a lot!!
